frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+// optional filter function for matching specific containers can be used.
+const getMountPaths = (podTemplate, /* optional function */ matchContainer) => {
+  const mountPaths = [];
+  const containers = _.get(podTemplate, 'spec.containers', []);
+  _.each(containers, function(container) {
+    if (matchContainer && !matchContainer(container)) {
+      return;
+    }
+
+    const mounts = _.get(container, 'volumeMounts', []);
+    _.each(mounts, function(mount) {
+      mountPaths.push(mount.mountPath);
+    });
+  });
+
+  return mountPaths;
+};
+
+//Return kind type give the plural
+const getKindFromPlural = plural => {
+  const ko = _.find(k8sModels, model => model.plural === plural);
+  return ko.kind;
+};
+
+const AttachStorageForm = () =>
+  class StorageFormComponent extends React.Component<
+    StorageProps,
+    StorageState
+    > {
+    constructor(props) {
+      super(props);
+      const existingResource = _.pick(props.obj, ['metadata', 'spec']);
+      const resourceObj = _.defaultsDeep({}, props.fixed, existingResource, {
+        apiVersion: 'v1',
+        data: {},
+        kind: props.kind,
+        metadata: {
+          name: '',
+        },
+        spec: {},
+      });
+
+      this.state = {
+        resourceObj,
+        inProgress: false,
+        formData: {},
+        readOnly: false,
+        disableForm: false,
+        checkboxItems: [
+          {
+            value: 'Mount as read-only',
+          },
+        ],
+        selectedOptions: [],
+      };
+    }
+
+    componentDidMount() {
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.plural === kind);
+      const supportedKinds = [
+        'Deployment',
+        'DeploymentConfig',
+        'ReplicaSet',
+        'ReplicationController',
+      ];
+
+      if (!_.includes(supportedKinds, ko.kind)) {
+        this.setState({
+          error: `Storage is not supported for kind ${ko.kind}.`,
+          inProgress: false,
+        });
+        return;
+      }
+
+      // Get the current resource so we can add to its definition
+      k8sGet(ko, metadata.name, metadata.namespace).then(
+        obj => {
+          this.setState({ resourceObj: obj });
+        },
+        err => this.setState({ error: err.message })
+      );
+    }
+
+    onError = error => {
+      this.setState({
+        error,
+        inProgress: false,
+      });
+    };
+
+    handleChange = event => {
+      const name = event.target.name;
+      const value = event.target.value;
+      const formData = { ...this.state.formData, [name]: value };
+      this.setState({ formData });
+    };
+
+    onReadOnlyChanged = event => {
+      let readOnly = this.state.readOnly;
+      readOnly = readOnly ? false : true;
+      this.setState({ readOnly });
+    };
+
+    onFormDisable = disable => {
+      this.setState({
+        disableForm: disable,
+      });
+    };
+
+    isValidForm = () => {
```

Comment:
`isValidForm` sounds like it is just checking the form, but it changes state, too. We should give a name like `validateForm` or something to make the side effect more clear.